new-frontiers-14/frontier-station-14
# Restore TV Audio

## About the PR


This PR restores the microphone components on wireless cameras (e.g. at The Pit, The Courthouse, and on Legman and Waveshot shuttles).  This means that televisions should have audio now.

Audio range for wireless cameras was reduced to 6 metres as an attempt to limit crowd noise on The Pit and Courthouse television channels.

This also adds a new ChatTransmitRange value as an attempt to prevent spamming admins (and only admins) with audio from out-of-range television sets.  As it stands, I believe this change also affects security camera monitors.  Not entirely happy with this, but it seems to reduce spam.

## Why / Balance

Watching courthouse or reporter feeds without audio was sad.  Running events like Crit in the Pit without television audio requires workarounds and ends up cluttering some radio channel unnecessarily.

Note: multiple televisions WILL send duplicate messages to anybody within earshot.  This could cause annoyance if multiple TVs were left behind inaccessible doors

## How to test

1. Build and run local server, start round.
2. Enter /aghost, warp to The Pit.
3. Spawn a wooden television set by the camera inside the glass box in the middle.
4. Tune the television into the Arena channel.
5. Spawn a Urist (dealer's choice), right-click on them, Debug, Control Mob.
6. Walk around the arena saying messages.  You should hear what you're saying through the television when between the north door and just shy of the south door, in a roughly 6 metre radius.
7. Run /aghost again.
8. Teleport to The Courthouse.
9. Spawn a wooden television set by the middle of the courthouse, and another in the deliberation room.
10. Tune in the television to the courthouse channel.
11. Spawn another Urist (dealer's choice), right click on them, Debug, Control Mob.
12. Walk around the Courtroom saying messages.  You should be able to hear yourself in all seats.  You should NOT hear yourself clearly when whispering in the jury's seats.  You should NOT hear yourself AT ALL in any of the seats in the jury's deliberation room.
13. Spawn a Vendor right beside the camera.
14. Check the View Variables for the Vendor.
15. In the Server Components, open the C.Se.AdvertiseComponent.  Set the MinimumWait to 3, the MaximumWait to 4, and the NextAdvertisementTime to a time a few seconds from now (check your PDA).
16. Wait for the vendor to start repeating messages.
17. /Run aghost again.
18. Wait around the television.  You should hear the vendor announcements through the TV.
19. Move until the television is out of sight.  You should NOT hear the vendor announcements through the TV.
20. Run /deadmin.  You should NOT hear the vendor announcements through the TV.
21. Return to the courtroom as a ghost so a television is in view.  You should hear the vendor announcements through the TV again.

## Media


The following images show the (approximate) respective ranges of the cameras at The Pit and The Courtroom.

![thepitgreencircle](https://github.com/new-frontiers-14/frontier-station-14/assets/166147148/49ef4881-2ca1-4ca8-bfce-14c72c7f609d)
![courtroomgreencircle](https://github.com/new-frontiers-14/frontier-station-14/assets/166147148/0905db52-8aef-4d29-a8f8-4e4c65a674d0)


## Breaking changes



**Changelog**

:cl:
- add: Televisions once again broadcast chatter from around the camera being watched.
